yolov5/train.py

```diff
@@ -24,96 +29,48 @@
+                           intersect_dicts, labels_to_class_weights, labels_to_image_weights, methods, one_cycle,
+                           print_args, print_mutation, strip_optimizer)
+from utils.loggers import Loggers
+from utils.loggers.wandb.wandb_utils import check_wandb_resume
+from utils.loss import ComputeLoss
+from utils.metrics import fitness
+from utils.plots import plot_evolve, plot_labels
+from utils.torch_utils import EarlyStopping, ModelEMA, de_parallel, select_device, torch_distributed_zero_first
 
-import yolov5.val as val  # for end-of-epoch mAP
-from yolov5.models.experimental import attempt_load
-from yolov5.models.yolo import Model
-from yolov5.utils.autoanchor import check_anchors
-from yolov5.utils.aws import upload_file_to_s3, upload_folder_to_s3
-from yolov5.utils.callbacks import Callbacks
-from yolov5.utils.datasets import create_dataloader
-from yolov5.utils.downloads import attempt_download
-from yolov5.utils.general import (check_dataset, check_file, check_git_status,
-                                  check_img_size, check_requirements,
-                                  check_suffix, check_yaml, colorstr,
-                                  get_latest_run, increment_path, init_seeds,
-                                  labels_to_class_weights,
-                                  labels_to_image_weights, methods, one_cycle,
-                                  print_args, print_mutation, set_logging,
-                                  strip_optimizer, yolov5_in_syspath)
-from yolov5.utils.loggers import Loggers
-from yolov5.utils.loggers.wandb.wandb_utils import check_wandb_resume
-from yolov5.utils.loss import ComputeLoss
-from yolov5.utils.metrics import fitness
-from yolov5.utils.plots import plot_evolve, plot_labels
-from yolov5.utils.torch_utils import (EarlyStopping, ModelEMA, de_parallel,
-                                      intersect_dicts, select_device,
-                                      torch_distributed_zero_first)
-
-LOGGER = logging.getLogger(__name__)
 LOCAL_RANK = int(os.getenv('LOCAL_RANK', -1))  # https://pytorch.org/docs/stable/elastic/run.html
 RANK = int(os.getenv('RANK', -1))
 WORLD_SIZE = int(os.getenv('WORLD_SIZE', 1))
 
-# fix OMP: Error #15: Initializing libiomp5md.dll, but found libiomp5md.dll already initialized.
-os.environ['KMP_DUPLICATE_LIB_OK']='True'
-
 
 def train(hyp,  # path/to/hyp.yaml or hyp dictionary
           opt,
           device,
           callbacks
           ):
-    save_dir, epochs, batch_size, weights, single_cls, evolve, data, cfg, resume, noval, nosave, workers, freeze, = \
+    save_dir, epochs, batch_size, weights, single_cls, evolve, data, cfg, resume, noval, nosave, workers, freeze = \
         Path(opt.save_dir), opt.epochs, opt.batch_size, opt.weights, opt.single_cls, opt.evolve, opt.data, opt.cfg, \
         opt.resume, opt.noval, opt.nosave, opt.workers, opt.freeze
 
-    # coco to yolov5 conversion
```

Comment:
this block is an important feature, you shouldnt remove it

Reply:
just to understand: this code is not in the original yolov5 code: is it an additional contribution you've made?
In that case, it would make sense to submit it upstream, no?

Reply:
i have submitted many prs to the upstream but they dont accept any of my features. thats why i implement many features in this package as aws, neptune and coco format support 👍

Reply:
ok, make sense. In that case, I understand why bumping the yolo version is more tricky than what I thought. Sorry for the wasted time!